### DISCUSSION TITLE
MSBuild GenerateFullPaths option and ProblemMatcher's fileLocation setting

### DISCUSSION THREAD

**Discussion by TryerGit** [microsoft/vscode-cpptools#8042] [Q&A]
Hello,

I use msbuild.exe called via an appropriately defined task in `tasks.json` the relevant part of which is:

```
"label": "winbuilddebug",
"type": "shell",
"command": "msbuild",
"args": [
               ".vscode/windows.vcxproj",
                "/property:GenerateFullPaths=true",
                "/property:Configuration=Debug",
                "/property:Platform=x64",
                "/t:build",
                "/consoleloggerparameters:NoSummary"
            ],
"group": "build",
"presentation": {
                // Reveal the output only if unrecognized errors occur.
                "reveal": "silent"
},
"problemMatcher":{ 
        "base": "$msCompile", 
//For some reason, even though above gives full path, in its display, providing
//relative below allows VSCode Problems tab to correctly go to this location
        "fileLocation": "relative"
//If "absolute" is specified for fileLocation, VSCode complains that it cannot find the file and gives option of
//creating a new file with that name
}
```
When the above task is run, I am able to observe in the Terminal tab as the compilation / build process proceeds that warnings / errors are issued by the compilation / build process. A sample of such a warning looks thus:

```
E:\project\src\file.cpp(409,53): warning C4267: 'argument': conversion from 'size_t' to
 'int', possible loss of data [E:\project\.vscode\windows.vcxproj]
```

On searching through what exactly GenerateFullPaths does in a build's output process, (see for instance https://docs.microsoft.com/en-us/visualstudio/msbuild/common-msbuild-project-properties?view=vs-2019), I suppose that the full paths in the warning above come about because GenerateFullPaths was set to true.

Now, this being the case, I would like to understand why "fileLocation" set to "relative" as opposed to "absolute" works. That is, when I set "fileLocation" to "absolute" with GenerateFullPaths set to true, VSCode complains that it is unable to open the file, File not found (which seems to have been the source of quite of bit of posts, see for instance https://github.com/microsoft/vscode/issues/50426, https://stackoverflow.com/questions/39306348/visual-studio-code-unable-to-open-myfile-h-file-not-found, https://stackoverflow.com/questions/60472084/vscode-unable-to-open-file-unable-to-read-file-message-when-clicking-on-an)

Yet, when this setting is set to "relative", it works -- that is, navigating to that error on the Problems tab takes me directly to the line in the source code (in the example above, line 409, column 53). Why does this work while "absolute" does not?

Thank you.

**Comment by sean-mcmanus** [comment]
I don't know either (I agree that it seems like a bug). Our C/C++ extension doesn't provide the $msbuild problemMatcher or the "shell" task and doesn't process the fileLocation property. You'd have to ask someone on the VS Code team: https://github.com/microsoft/vscode/issues
